keras/backend/theano_backend.py

```diff
@@ -324,8 +325,11 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, std, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, std, beta and gamma.
     '''
-    normed = T.nnet.bn.batch_normalization(x, gamma, beta, mean, std + epsilon,
-                                           mode='high_mem')
+    if theano.config.device.startswith('gpu'):
+        normed = dnn_batch_normalization_test(x, gamma, beta, mean, std ** 2, epsilon)
```

Comment:
Wy `**2`?

Have you look at the `mode` argument? Seem like it's more complicated than that.